### PR TITLE
✨ Update iscn-dev widget to fotan

### PR DESCRIPTION
**Pull request by williamchong**

### CODE REVIEW

File: pages/in/widget/iscn/index.vue [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+        const {
+          fingerprint,
+          title,
+          tags,
+          type,
+          license,
+          publisher,
+          url,
+        } = this;
+        const txHash = await this.sendISCNSignature({
```
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:310]: @williamchong here need to add a signer to pass in:
before const txHash:
const signer = await this.prepareCosmosTxSigner(); // add this line
const txHash = await this.sendISCNSignature({
          userId: this.getUserId,
          ...
          signer, // add this line
 })

File: util/cosmos/iscn/constant.js [hunk 2]
```diff
@@ -15,3 +15,5 @@ export const ISCN_LICENSES = {
     '/': 'QmZhRNkZaSnhDr6gBC22zwhTjsGyUx39tm8gjFYnTr2SjN',
   },
 };
+
+export const ISCN_RPC_URL = '/api/cosmos/rpc';
```
Comment (shellyYG) [inline util/cosmos/iscn/constant.js:19]: need to add 
import { EXTERNAL_URL } from '@/constant';
...
export const ISCN_RPC_URL = `${EXTERNAL_URL}/api/cosmos/rpc`;

File: util/cosmos/iscn/query.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+import { Tendermint34Client } from '@cosmjs/tendermint-rpc';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { decodeTxRaw } from '@cosmjs/proto-signing';
+import { QueryClient, StargateClient } from '@cosmjs/stargate';
+import { MsgCreateIscnRecord } from '@likecoin/iscn-message-types/dist/iscn/tx';
+
+import setupISCNExtension from './iscnQueryExtension';
+import { timeout } from '@/util/misc';
+
+const ISCN_RPC_URL = '/api/cosmos/rpc';
```
Comment (shellyYG) [inline util/cosmos/iscn/query.js:11]: need to add:
import { EXTERNAL_URL } from '@/constant';
const ISCN_RPC_URL = `${EXTERNAL_URL}/api/cosmos/rpc`;

**Review by shellyYG — CHANGES_REQUESTED** [review]

File: pages/in/widget/iscn/index.vue [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+  },
+  methods: {
+    ...mapActions([
+      'popupAuthDialogInPlace',
+      'setReAuthDialogShow',
+      'doUserAuth',
+      'sendISCNSignature',
+      'setErrorMsg',
+      'closeTxDialog',
+      'prepareCosmosTxSigner',
```
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:292]: add `fetchCurrentCosmosWallet`

File: pages/in/widget/iscn/index.vue [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+
+        const {
+          fingerprint,
+          title,
+          tags,
+          type,
+          license,
+          publisher,
+          url,
+        } = this;
```
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:309]: after } = this; add
`const from = await this.fetchCurrentCosmosWallet();
        if (!from) {
          throw new Error('VALIDATION_FAIL');
        }
        const userWallet = cosmosWallet;
        if (from !== userWallet) {
          this.setErrorMsg(this.$t('Transaction.error.authcoreWalletNotMatch'));
          throw new Error('VALIDATION_FAIL');
        }
        const signer = await this.prepareCosmosTxSigner()`

File: pages/in/widget/iscn/index.vue [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+        const txHash = await this.sendISCNSignature({
+          userId: this.getUserId,
+          displayName: this.getUserInfo.displayName,
+          cosmosWallet,
```
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:313]: cosmosWallet -> change to `from`

File: pages/in/widget/iscn/index.vue [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+          fingerprint,
+          title,
+          tags,
+          type,
+          license,
+          publisher,
+          url,
+          showDialogAction,
+          isWait,
```
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:322]: after isWait, add `signer,`

File: store/modules/actions/payment.js [hunk 8]
```diff
@@ -82,7 +83,7 @@ export async function sendISCNSignature(
       publisher,
       memo,
     } = payload;
-    const { txHash, included } = await remoteSignISCNPayload({
+    const { transactionHash } = await signISCNTx({
```
Comment (shellyYG) [inline store/modules/actions/payment.js:86]: above
`const {
      userId,
      displayName,
      cosmosWallet,
      fingerprint,
      title,
      tags,
      type,
      license,
      publisher,
      memo,
    } = payload;`
    
    need to replace 'cosmosWallet' with 'from'
`const {
      userId,
      displayName,
      from,
      fingerprint,
      title,
      tags,
      type,
      license,
      publisher,
      memo,
    } = payload;`

File: store/modules/actions/payment.js [hunk 9]
```diff
@@ -82,7 +83,7 @@ export async function sendISCNSignature(
...
       userId,
       displayName,
       cosmosWallet,
```
Comment (shellyYG) [inline store/modules/actions/payment.js:89]: `cosmosWallet`--change to-->`from`

File: store/modules/actions/payment.js [hunk 10]
```diff
@@ -92,14 +93,13 @@ export async function sendISCNSignature(
       type,
       license,
       publisher,
-      memo,
-    });
+    }, signer, cosmosWallet, memo);
```
Comment (shellyYG) [inline store/modules/actions/payment.js:96]: `cosmosWallet` --change to-->`from`

**Review by shellyYG — CHANGES_REQUESTED** [review]

File: store/modules/actions/payment.js (hunk 10 above)
Comment (shellyYG) [inline store/modules/actions/payment.js:96]:
```suggestion
    }, signer, from, memo);
```

File: util/cosmos/iscn/query.js (hunk 3 above)
Comment (shellyYG) [inline util/cosmos/iscn/query.js:11]:
```suggestion
const ISCN_RPC_URL = `${EXTERNAL_URL}/api/cosmos/rpc`;
```

File: pages/in/widget/iscn/index.vue (hunk 5 above)
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:309]:
```suggestion
        } = this;
       const from = await this.fetchCurrentCosmosWallet();
        if (!from) {
          throw new Error('VALIDATION_FAIL');
        }
        const userWallet = cosmosWallet;
        if (from !== userWallet) {
          this.setErrorMsg(this.$t('Transaction.error.authcoreWalletNotMatch'));
          throw new Error('VALIDATION_FAIL');
        }
        const signer = await this.prepareCosmosTxSigner();
```

File: pages/in/widget/iscn/index.vue (hunk 6 above)
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:313]:
```suggestion
          from,
```

File: pages/in/widget/iscn/index.vue (hunk 7 above)
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:322]:
```suggestion
          isWait,
          signer,
```

File: pages/in/widget/iscn/index.vue (hunk 4 above)
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:292]:
```suggestion
      'fetchCurrentCosmosWallet',
      'prepareCosmosTxSigner',
```

File: util/cosmos/iscn/query.js (hunk 3 above)
Comment (shellyYG) [inline util/cosmos/iscn/query.js:11]:
```suggestion
const ISCN_RPC_URL = `${EXTERNAL_URL}/api/cosmos/rpc`;
```

File: util/cosmos/iscn/constant.js (hunk 2 above)
Comment (shellyYG) [inline util/cosmos/iscn/constant.js:19]:
```suggestion
export const ISCN_RPC_URL = `${EXTERNAL_URL}/api/cosmos/rpc`;
```

File: pages/in/widget/iscn/index.vue [hunk 11]
```diff
@@ -0,0 +1,402 @@
...
+        // if (this.redirectUri) {
+        //   const url = new URL(this.redirectUri, true);
+        //   url.set('query', { ...url.query, ...payload });
+        //   window.location.href = url.toString();
+        // }
+      } else if (this.getIsShowingTxPopup) {
+        this.closeTxDialog();
+        // TODO: swap to iscn page
+        this.$router.push({
+          name: 'in-tx-iscn-dev-id',
```
Comment (shellyYG) [inline pages/in/widget/iscn/index.vue:361]:
```suggestion
          name: 'in-tx-iscn-id',
```

File: util/cosmos/iscn/query.js (hunk 3 above)
Comment (williamchong) [reply util/cosmos/iscn/query.js]: Refactored to import `ISCN_RPC_URL` from `./constant`

File: pages/in/tx/iscn/_id/index.vue [hunk 12]
```diff
@@ -0,0 +1,605 @@
...
+            </ul>
+          </div>
+        </section>
+      </div>
+    </template>
+
+    <footer class="iscn-panel__footer">
+      <a
+        class="iscn-block-button"
+        :href="`https://node.iscn-dev.like.co/txs/${txId}`"
```
Comment (shellyYG) [inline pages/in/tx/iscn/_id/index.vue:305]:
```suggestion
        :href="`https://node.iscn-dev-2.like.co/txs/${txId}`"
```

**Pull request merged**